packages/inertia/src/router.ts

```diff
@@ -360,18 +363,24 @@ export class Router {
     replace = false,
     preserveScroll = false,
     preserveState = false,
+    preserveUrl = false,
   }: {
     visitId?: VisitId,
     replace?: boolean,
     preserveScroll?: PreserveStateOption
     preserveState?: PreserveStateOption
+    preserveUrl?: PreserveStateOption
   } = {}): Promise<void> {
     return Promise.resolve(this.resolveComponent(page.component)).then(component => {
       if (visitId === this.visitId) {
         page.scrollRegions = page.scrollRegions || []
         page.rememberedState = page.rememberedState || {}
         replace = replace || hrefToUrl(page.url).href === window.location.href
-        replace ? this.replaceState(page) : this.pushState(page)
+
+        if (! preserveUrl) {
+          replace ? this.replaceState(page) : this.pushState(page)
```

Comment:
Hey @xiCO2k 
the current page object is also set inside the `replaceState` and `pushState` methods which could introduce side effects. What you probably want to avoid is the `window.history` inside those methods?

Reply:
First I did exactly that, and set the `the.page = page` also with the `preserveUrl`, but actually does not work well.

When you load the `page=2` it does not change the url on top but when you load the `page=3` it changes the url to `page=2`, its related to the `handlePopstateEvent` method.